elastic/kibana
# [Entity Analytics] Add risk engine missing privileges callout

## Summary

_note: this is currently behind the experimental feature flag `riskEnginePrivilegesRouteEnabled`._

Add a callout to the Entity Risk Score Management page if the user doesn't have sufficient privileges. Here is the callout with a user with none of the privileges (missing privileges are dynamically shown)

<img width="1177" alt="Screenshot 2023-11-21 at 12 52 21" src="https://github.com/elastic/kibana/assets/3315046/0c4a17ee-8856-45a5-8798-1cef0e7fe0ad">

as part of this I have added a route `GET /internal/risk_score/engine/privileges` the response payload looks like this:
```
{
    "privileges": {
        "kibana": {
            "feature_savedObjectsManagement.all": false
        },
        "elasticsearch": {
            "cluster": {
                "manage_transform": false,
                "manage_index_templates": false
            },
            "index": {
                "risk-score.risk-score-*": {
                    "read": false,
                    "write": false
                }
            }
        }
    },
    "has_all_required": false // does the user have all privileges? 
}
```

Docs issue for associated documentation changes https://github.com/elastic/security-docs/issues/4307

### Testing    
- cypress tests added for the no banner case (user has all privs), and the worst case (user has none of the privs)
- API Integration tests added for all of the granular cases 
- Manual test steps
    - 1.  User has correct privileges
        - Create a user with all risk engine privileges
        - navigate to the Entity Risk Score Management page
        - missing privileges banner should not show
    - 2.  User has missing privileges
        - Create a user with some or no risk engine privileges
        - navigate to the Entity Risk Score Management page
        - banner should show and describe all privileges missing

### Checklist

Delete any items that are not applicable to this PR.
